### DISCUSSION TITLE
UpdateInput allow modify ID

### DISCUSSION THREAD

**Discussion by ismaserrano** [prisma/prisma#18031] [Q&A]
Why if I send `data: { id: newID, ... }` to an a existing record, data is typed by `*UpdateInput` and `id` is not in type definition, `db.entity.update({data})` allows change the record ID?

**Comment by nurul3101** [comment]
Hey @ismaserrano 👋 

Can you share the schema file or the entity model and the query you are invoking to understand better?
I am not able to fully understand your question.

**Reply by ismaserrano** [reply]
Thanks for reply @nurul3101 ;)

```
model Event {
  id           Int         @id @unique @default(autoincrement())
  name         String      @db.VarChar(200)
  description  String      @db.VarChar(20000)
  slug         String      @unique @db.VarChar(2000)

  createdAt       DateTime @default(now())
  updatedAt       DateTime @updatedAt
  scheduledAt     DateTime @default(now())
  timezone        String   @default("Europe/Madrid") @db.VarChar(200)

  // Relation fields
  owner   User @relation(fields: [ownerId], references: [id])
  ownerId Int
}
```

And the query:
```
public async update(id: number, data: Prisma.EventUpdateInput): Promise<void> {
.
.
.
event = await db.event.update({
          data,
          where: {
            id: id,
          },
        });
.
.
.
}
```

The export type generated by Prisma based in schema:

```
export type EventUpdateInput = {
    name?: StringFieldUpdateOperationsInput | string
    description?: StringFieldUpdateOperationsInput | string
    slug?: StringFieldUpdateOperationsInput | string
    createdAt?: DateTimeFieldUpdateOperationsInput | Date | string
    updatedAt?: DateTimeFieldUpdateOperationsInput | Date | string
    scheduledAt?: DateTimeFieldUpdateOperationsInput | Date | string
    timezone?: StringFieldUpdateOperationsInput | string
    owner?: UserUpdateOneRequiredWithoutEventNestedInput
  }
```

**Reply by nurul3101** [reply]
Hey @ismaserrano 
There are two input types, `EventUpdateInput` and `EventUncheckedUpdateInput`. EventUncheckedUpdateInput has the id field while EventUpdateInput doesn't have it.
<img width="645" alt="Screenshot 2023-02-24 at 8 41 21 PM" src="https://user-images.githubusercontent.com/13049676/221214350-b06b7938-907c-40b8-913e-86767f3f0197.png">

**Reply by ismaserrano** [reply]
Exactly, you’re right. If you see data in function, is `EventUpdateInput` which doesn’t have `id` in the type… thats’s the problem. Then, why the `id` can be modified? That’s no sense! 😅

**Reply by nurul3101** [reply]
Can you let me know your main use case? Would you like to prevent `id` from being updated?

**Reply by ismaserrano** [reply]
Hi @nurul3101, thank you for your time. I hope to explain better this time.

If I've got this `data` object:
```
data: {
  name: 'New name',
  description: 'New description',
}
```
The `update` function (which is called via PATCH method to`/api/events/oldID`) updates this data correctly, but if I put:
```
data: {
  id: 'newID',
  name: 'New name',
  description: 'New description',
}
```
with the same call, the `id` is updated to `newID` and that can't be possible! Because `id` is not in `EventUpdateInput` type.
Yes, I'd like to prevent `id` being updated but `delete data.id` is not the better way to do that.
This is for security purposes... not use case.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
